homeassistant/components/roborock/sensor.py

```diff
@@ -65,7 +65,9 @@ def _dock_error_value_fn(properties: DeviceProp) -> str | None:
         icon="mdi:brush",
         device_class=SensorDeviceClass.DURATION,
         translation_key="main_brush_time_left",
-        value_fn=lambda data: data.consumable.main_brush_time_left,
+        value_fn=lambda data: data.consumable.main_brush_time_left
+        if data.consumable is not None
+        else None,
```

Comment:
Why aren't you using something like `getattr(data.consumable, 'main_brush_time_left',None)`?

Reply:
I was under the impression getter was preferably avoided

Reply:
A few of my None checks were actually not needed, so there are just two of them left I believe. If you would like I can change to getattr, but I personally think the None check is a little easier to follow and we don't lose any typing, But I am happy to change if you'd like

Reply:
No you can leave it as there are only two :)